In [33]:
import math
import pandas as pd
from scipy.stats import t

In [2]:
df = pd.DataFrame({
    "Pesos RN": {
        0: 2.652552032,
        1: 2.761094312,
        2: 2.709140337,
        3: 3.022067451,
        4: 2.576154022,
        5: 2.857961865,
        6: 2.913367372,
        7: 3.382368324,
        8: 1.72839582,
        9: 2.298327037,
        10: 3.031230155,
        11: 2.535162186,
        12: 3.427837602,
        13: 2.49240775,
        14: 2.683317054,
        15: 2.683240656,
        16: 3.287451638,
        17: 3.665004717
    }
})

In [10]:
df.describe()

Pesos RN
count  18.000000
mean    2.817060
std     0.454167
min     1.728396
25%     2.595254
50%     2.735117
75%     3.028939
max     3.665005

In [26]:
alpha = 0.05
h0 = 2.5

In [6]:
pesos_rn = df["Pesos RN"]

In [25]:
size_rn = len(pesos_rn)
mean_rn = pesos_rn.mean()
std_rn = pesos_rn.std()
gl = size_rn - 1

Se cree que los pesos de los recién nacidos son superiores a 2.5 kg en promedio.  
#### Pasos:
1. Datos  
   Los datos provienen de la recolección de los pesos a nacer de bebes. Nosotros queremos estudiar el peso promedio de los recién nacidos con un $\alpha=0.05$
2. Supuestos
   Los pesos de los RN siguen una distribución normal. Se desconoce la desviación estándar y se tiene una muestra pequeña ($n=18$)
3. Hipótesis
   Hipótesis de investigación: El peso promedio de RN es superior a 2.5 kg  
   $H_A > 2.5$kg vs $H_0 \le 2.5$kg
4. Estadístico de Prueba
   $$T=\frac{\bar{x} - \mu_0}{\frac{s}{\sqrt n}}$$
   $$s^2 = \frac{1}{n-1}\sum_{i=1}^{k}(x_i-\bar{x})^2$$
5. Distribución del estadístico de prueba
   $$T \sim t_{n-1}$$
6. Región de rechazo

In [40]:
# Valor crítico
vc_superior = t.ppf(1 - alpha, gl) # Usa sola cola superior, en caso de dos usar 1 - alpha / 2
vc_superior

1.7396067260750672

7. Evaluar el estadístico

In [34]:
t_val = (mean_rn - h0) / (std_rn / math.sqrt(size_rn))
t_val

2.9618407952201258

8. Decisión  
   Como $t > VC_{Superior}$ este pertenece a la zona de rechazo,  
   Por lo que se decide rechazar $H_0$
9. Conclusión  
    Encontramos que el promedio de los 18 recién nacidos es superior a 2.5 kg
10. p-value  
    $p_{value} = P(t_{17} > 2.96)$

In [38]:
p_value = 1 - t.cdf(t_val, 17)
p_value

0.004368311007374115

$p_{value} < \alpha = 0.05$